src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetDeleteInstancesMethod.cs

```diff
@@ -50,13 +50,27 @@ public override void ExecuteCmdlet()
                     System.Collections.Generic.IList<string> instanceIds = this.InstanceId;
 
                     Rest.Azure.AzureOperationResponse result = null;
-                    if (!string.IsNullOrEmpty(resourceGroupName) && !string.IsNullOrEmpty(vmScaleSetName) && instanceIds != null)
+                    if (NoWait.IsPresent)
                     {
-                        result = VirtualMachineScaleSetsClient.DeleteInstancesWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        if (!string.IsNullOrEmpty(resourceGroupName) && !string.IsNullOrEmpty(vmScaleSetName) && instanceIds != null)
+                        {
+                            result = VirtualMachineScaleSetsClient.BeginDeleteInstancesWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        }
+                        else
+                        {
+                            result = VirtualMachineScaleSetsClient.BeginDeleteWithHttpMessagesAsync(resourceGroupName, vmScaleSetName).GetAwaiter().GetResult();
+                        }
                     }
                     else
                     {
-                        result = VirtualMachineScaleSetsClient.DeleteWithHttpMessagesAsync(resourceGroupName, vmScaleSetName).GetAwaiter().GetResult();
+                        if (!string.IsNullOrEmpty(resourceGroupName) && !string.IsNullOrEmpty(vmScaleSetName) && instanceIds != null)
+                        {
+                            result = VirtualMachineScaleSetsClient.DeleteInstancesWithHttpMessagesAsync(resourceGroupName, vmScaleSetName, instanceIds).GetAwaiter().GetResult();
+                        }
+                        else
+                        {
+                            result = VirtualMachineScaleSetsClient.DeleteWithHttpMessagesAsync(resourceGroupName, vmScaleSetName).GetAwaiter().GetResult();
+                        }
```

Comment:
same comment

Reply:
Note that, for all of these, if we just ensure that this GetOperationId method doesn't throw an exception when using with the Begin-* method, then we are OK - otherwise, we just may not want to do this step.

Reply:
I would say that we should do one automated test of one of these cmdlets that use thsi pattern (Get GetOperationIdFromUrlString as part of processing the output of the Begin* method).

Reply:
Agreed. I am writing down which are these methods. I will revert those for now and then write tests later.